lib/core/CJsonLogLayout.cc

```diff
@@ -63,132 +59,103 @@ bool CJsonLogLayout::locationInfo() const {
-    if (StringHelper::equalsIgnoreCase(option, LOG4CXX_STR("LOCATIONINFO"),
-                                       LOG4CXX_STR("locationinfo"))) {
-        this->locationInfo(OptionConverter::toBoolean(value, false));
-    }
-    if (StringHelper::equalsIgnoreCase(option, LOG4CXX_STR("PROPERTIES"),
-                                       LOG4CXX_STR("properties"))) {
-        this->properties(OptionConverter::toBoolean(value, false));
-    }
-}
-
-void CJsonLogLayout::format(LogString& output, const spi::LoggingEventPtr& event, Pool& /*p*/) const {
+void CJsonLogLayout::operator()(const boost::log::record_view& rec,
+                                boost::log::formatting_ostream& strm) const {
     using TStringBufferWriter = rapidjson::Writer<rapidjson::StringBuffer>;
     rapidjson::StringBuffer buffer;
-    TStringBufferWriter writer(buffer);
+    TStringBufferWriter writer{buffer};
 
     writer.StartObject();
 
     writer.String(LOGGER_NAME);
-    LOG4CXX_ENCODE_CHAR(logger, event->getLoggerName());
-    writer.String(logger);
+    writer.String(LOGGER);
 
     writer.String(TIMESTAMP_NAME);
-    writer.Int64(event->getTimeStamp() / 1000);
+    const boost::posix_time::ptime& timeStamp{
+        boost::log::extract<boost::posix_time::ptime>(
+            boost::log::aux::default_attribute_names::timestamp(), rec)
+            .get()};
+    writer.Int64((timeStamp - EPOCH).total_milliseconds());
 
     writer.String(LEVEL_NAME);
-    LOG4CXX_ENCODE_CHAR(level, event->getLevel()->toString());
-    writer.String(level);
+    CLogger::ELevel level{boost::log::extract<CLogger::ELevel>(
+                              boost::log::aux::default_attribute_names::severity(), rec)
+                              .get()};
+    writer.String(CLogger::levelToString(level));
 
     writer.String(PID_NAME);
     writer.Int64(PID);
 
     writer.String(THREAD_NAME);
-    LOG4CXX_ENCODE_CHAR(thread, event->getThreadName());
-    writer.String(thread);
+    const boost::log::attributes::current_thread_id::value_type threadId{
+        boost::log::extract<boost::log::attributes::current_thread_id::value_type>(
+            boost::log::aux::default_attribute_names::thread_id(), rec)
+            .get()};
+    std::ostringstream oss;
+    oss << threadId;
+    writer.String(oss.str());
 
     writer.String(MESSAGE_NAME);
-    LOG4CXX_ENCODE_CHAR(message, event->getRenderedMessage());
+    const std::string& message{rec[boost::log::expressions::smessage].get()};
     writer.String(message);
 
-    LogString logNdc;
-    if (event->getNDC(logNdc)) {
-        writer.String(NDC_NAME);
-        LOG4CXX_ENCODE_CHAR(ndc, logNdc);
-        writer.String(ndc);
-    }
-
     if (m_LocationInfo) {
-        const spi::LocationInfo& locInfo = event->getLocationInformation();
 
-        const std::string& className = locInfo.getClassName();
-        if (!className.empty()) {
+        std::string functionName{boost::log::extract<std::string>(
```

Comment:
suggestion: pull out the functionName processing into a separate function - it's probably worthy of a unit test too